presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    @Override
+    public void checkCanInsertIntoTable(Identity identity, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyInsertTable(table.toString());
+        }
+    }
+
+    @Override
+    public void checkCanDeleteFromTable(Identity identity, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyDeleteTable(table.toString());
+        }
+    }
+
+    @Override
+    public void checkCanCreateView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denyCreateView(view.toString());
+        }
+    }
+
+    @Override
+    public void checkCanDropView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denyDropView(view.toString());
+        }
+    }
+
+    @Override
+    public void checkCanSelectFromView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denySelectView(view.toString());
+        }
+    }
+
+    @Override
+    public void checkCanCreateViewWithSelectFromTable(Identity identity, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyCreateViewWithSelect(table.toString());
+        }
+    }
+
+    @Override
+    public void checkCanCreateViewWithSelectFromView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denyCreateViewWithSelect(view.toString());
+        }
+    }
+
+    @Override
+    public void checkCanSetCatalogSessionProperty(Identity identity, String catalogName, String propertyName)
+    {
+        if (!canAccessCatalog(identity, catalogName)) {
+            denySetCatalogSessionProperty(catalogName, propertyName);
+        }
+    }
+
+    @Override
+    public void checkCanGrantTablePrivilege(Identity identity, Privilege privilege, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyGrantTablePrivilege(privilege.name(), table.toString());
+        }
+    }
+
+    @Override
+    public void checkCanRevokeTablePrivilege(Identity identity, Privilege privilege, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyRevokeTablePrivilege(privilege.name(), table.toString());
+        }
+    }
+
+    private boolean canAccessCatalog(Identity identity, String catalogName)
+    {
+        for (CatalogAccessControlRule rule : catalogRules) {
+            Optional<Boolean> allowed = rule.match(identity.getUser(), catalogName);
+            if (allowed.isPresent()) {
+                return allowed.get();
+            }
+        }
+        return false;
```

Comment:
yes, the default is to deny access to all. That's how the file based access control is implemented too. Yeah, the documentation should make this clear.